platform/core/src/classes/ImageSet.ts

```diff
@@ -79,63 +75,96 @@ class ImageSet {
-      referenceImagePositionPatient[0],
-      referenceImagePositionPatient[1],
-      referenceImagePositionPatient[2]
-    );
-
-    const ImageOrientationPatient = _getImageOrientationPatient(images[0]);
-
-    const scanAxisNormal = new Vector3(
-      ImageOrientationPatient[0],
-      ImageOrientationPatient[1],
-      ImageOrientationPatient[2]
-    ).cross(
-      new Vector3(
-        ImageOrientationPatient[3],
-        ImageOrientationPatient[4],
-        ImageOrientationPatient[5]
-      )
-    );
-
-    const distanceImagePairs = images.map(function (image: Image) {
-      const ippVec = new Vector3(..._getImagePositionPatient(image));
-      const positionVector = refIppVec.clone().sub(ippVec);
-      const distance = positionVector.dot(scanAxisNormal);
-
-      return {
-        distance,
-        image,
-      };
+  /**
+   * Sort by InstanceNumber
+   * Note: Images are sorted in-place and a reference to the sorted image array is returned.
+   *
+   * @returns images - reference to images after sorting
+   */
+  sortByInstanceNumber(): Image[] {
+    return this.sortBy((a, b) => {
+      // Sort by InstanceNumber (0020,0013)
+      // @ts-ignore
+      return (parseInt(a.InstanceNumber) || 0) - (parseInt(b.InstanceNumber) || 0);
     });
+  }
 
-    distanceImagePairs.sort(function (a, b) {
-      return b.distance - a.distance;
-    });
+  /**
+   * Sort by image position, calculated using ImageOrientationPatient and ImagePositionPatient
+   * Note: Images are sorted in-place and a reference to the sorted image array is returned.
+   *
+   * @returns images - reference to images after sorting
+   */
+  sortByImagePositionPatient(): Image[] {
+    const images = this.images;
 
-    const sortedImages = distanceImagePairs.map(a => a.image);
+    if (images.length <= 1) {
+      return; // No need to sort if there's only one image
+    }
 
-    images.sort(function (a, b) {
-      return sortedImages.indexOf(a) - sortedImages.indexOf(b);
-    });
-  }
-}
+    // Use the first image as a reference
+    const referenceImagePositionPatient = images[0].ImagePositionPatient;
+    const ImageOrientationPatient = images[0].ImageOrientationPatient;
+
+    if (!referenceImagePositionPatient) {
+      throw new Error(
+        'Cannot sort ImageSet by real-world positions - ImagePositionPatient is undefined'
+      );
+    } else if (!ImageOrientationPatient) {
+      throw new Error(
+        'Cannot sort ImageSet by real-world positions - ImageOrientationPatient is undefined'
+      );
+    }
 
-function _getImagePositionPatient(image) {
-  return image.getData().metadata.ImagePositionPatient;
-}
+    // Calculate the scan axis normal using the cross product
+    const scanAxisNormal = calculateScanAxisNormal(ImageOrientationPatient);
+
+    // Compute distances from each image to the reference image
+    const distanceInstancePairs = images.map(image => {
+      const imagePositionPatient = image.ImagePositionPatient;
```

Comment:
Please use the 
const deltaVector = vec3.create();
...
vec3.dot(scanAxisNormal, vec3.subtract(deltaVector,imagePositionPatient, referenceImagePositionPatient)) to calculate distances rather than re-implementing it dot product using reduce.  It makes it clearer what is going on.